File: multiscreen/app/src/main/java/com/example/multiscreen/MainActivity.java
package com.example.multiscreen;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    TextView name;
    Button button;

    public static final String EXTRA_MESSAGE = "com.example.multiscreen.EXTRA_MESSAGE";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

    }
    public void openActivity(View view) {

        Intent intent = new Intent(this, MainActivity2.class);
        name = findViewById(R.id.name);
        button = findViewById(R.id.button);
        String name1 = name.getText().toString();
        intent.putExtra(EXTRA_MESSAGE, name1);
        startActivity(intent);

    }
}

File: kotlinMultiscreen/app/src/main/java/com/example/kotlinmultiscreen/MainActivity.kt
package com.example.kotlinmultiscreen

import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        var createButton = findViewById<Button>(R.id.button)
        var createEditText = findViewById<EditText>(R.id.editTextText)
    }

    fun openActivity(view: View) {
        val editText = findViewById<EditText>(R.id.editTextText)
        val message = editText.text.toString()
        val intent = Intent(this, SecondActivity::class.java).apply {
            putExtra("EXTRA_MESSAGE", message)
        }
        startActivity(intent)

        
    }
}